File: network/src/main/java/app/allever/android/lib/network/ResultThrowable.java
package app.allever.android.lib.network;

/**
 * @ClassName RsThrowable
 * @Description TODO
 * @Author Jerry
 * @Date 2020/5/4 0:17
 * @Version 1.0
 */
public class ResultThrowable extends Exception {

    public int code;
    public String message;

    public ResultThrowable(Throwable cause, int code) {
        super(cause);
        this.code = code;
    }

    public ResultThrowable(int code, String message) {
        this.code = code;
        this.message = message;
    }
}

File: network/src/main/java/app/allever/android/lib/network/ResultThrowable.kt
package app.allever.android.lib.network

import java.lang.Exception

/**
 * @ClassName RsThrowable
 * @Description TODO
 * @Author Jerry
 * @Date 2020/5/4 0:17
 * @Version 1.0
 */
class ResultThrowable : Exception {
    var code: Int
    override var message: String? = null

    constructor(cause: Throwable?, code: Int) : super(cause) {
        this.code = code
    }

    constructor(code: Int, message: String?) {
        this.code = code
        this.message = message
    }
}